chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -103,6 +116,12 @@ pub(crate) struct NetworkState {
     /// A graph of the whole NEAR network.
     pub graph: Arc<crate::routing::Graph>,
 
+    /// Information about recent outbound connections which lasted for at least
+    /// RECENT_OUTBOUND_CONNECTIONS_MIN_DURATION
+    pub recent_outbound_connections: Arc<Mutex<Vec<ConnectionInfo>>>,
+    /// Number of remaining attempts to re-establish a recent outbound connection
+    pub pending_reconnect_attempts: Arc<Mutex<HashMap<PeerId, (PeerInfo, usize)>>>,
+
```

Comment:
how about putting these 2 fields into a separate struct with will expose a bunch of atomic operations?

Reply:
NetworkState is large and each functionality that can be isolated from it deserves to be a separate component with a narrow API.